### PR TITLE
Close #11 transactions relationship endpoints

### LINKED ISSUE #11: Transactions relationship endpoints

### PR DESCRIPTION
**Pull request by cconcannon**

### CODE REVIEW

File: spec/controllers/merchants_invoices_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+RSpec.describe Api::V1::Merchants::InvoicesController do
+  describe "GET all related invoices" do
+    it "retrieves all invoices related to that merchant" do
+      merchant = FactoryGirl.create(:merchant, id: 1)
+      10.times do
+        FactoryGirl.create(:invoice, merchant: merchant)
+      end
+      
+      get :index, params: { format: :json, id: 1 }
```
Comment (GSmes) [inline spec/controllers/merchants_invoices_controller_spec.rb:11]: @concach 
Now that we added that line in ApplicationController to make JSON the default format, do we still need to pass the format in the params here?

Also on line 22 of this file.

File: spec/controllers/merchants_items_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::Merchants::ItemsController do
+  describe "GET related items for that merchant" do
+    it "retrieves items related to that merchant" do
+      merchant = FactoryGirl.create(:merchant, id: 1)
+      5.times do
+        FactoryGirl.create(:item, merchant: merchant)
+      end
+      get :index, params: {
```
Comment (GSmes) [inline spec/controllers/merchants_items_controller_spec.rb:10]: @concach 
See my comment in spec/controllers/merchants_invoices_controller_spec.rb.

Also on line 22 of this file.

File: spec/controllers/transactions_invoice_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    it "retrieves the correct invoice for that transaction" do
+      transaction = FactoryGirl.create(:transaction,      
+        id: 1,
+        invoice: FactoryGirl.create(:invoice, id: 20)
+      )
+      10.times do
+        FactoryGirl.create(:invoice)
+      end
+      
+      get :show, params: { format: :json, id: 1 }
```
Comment (GSmes) [inline spec/controllers/transactions_invoice_controller_spec.rb:14]: @concach 
See my comment in spec/controllers/merchants_invoices_controller_spec.rb.

Also on line 28 of this file.

**Comment by GSmes** [comment]
@concach 
Great work! And fast!
See my comments in the files. I didn't check your previous PR to see if the same changes can be made, so if you decide to make the changes, I would go through the files in that PR, too

**Comment by cconcannon** [comment]
good point - I just tested with my most recent branch and you're right. I'll make the changes there and submit a new PR

**Pull request closed without merging**